treeherder/webapp/api/performance_data.py

```diff
@@ -423,3 +427,68 @@ class PerformanceIssueTrackerViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = IssueTrackerSerializer
     filter_backends = [filters.OrderingFilter]
     ordering = 'id'
+
+
+class PerformanceByRevision(generics.ListAPIView):
+
+    serializer_class = PerformanceRevisionSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = PerformanceQueryParamsSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors,
+                            status=HTTP_400_BAD_REQUEST)
+
+        startday = query_params.validated_data['startday']
+        endday = query_params.validated_data['endday']
+        revision = query_params.validated_data['revision']
+        repository = query_params.validated_data['repository']
+        interval = query_params.validated_data['interval']
+        frameworks = query_params.validated_data['framework']
+        parent_signature = query_params.validated_data['parent_signature']
+        no_subtests = query_params.validated_data['no_subtests']
+
+        signature_data = (PerformanceSignature.objects
+                                              .select_related('framework', 'repository', 'platform', 'push')
+                                              .filter(repository__name=repository, framework__in=frameworks,
+                                                      parent_signature__isnull=no_subtests))
+        if parent_signature:
+            signature_data = signature_data.filter(parent_signature__signature_hash=parent_signature)
+
+        if interval:
+            signature_data = signature_data.filter(last_updated__gte=datetime.datetime.utcfromtimestamp(
+                                                   int(time.time() - int(interval))))
+
+        self.queryset = (signature_data.values('framework_id', 'id', 'lower_is_better', 'has_subtests', 'extra_options',
+                                               'suite', 'platform__platform', 'test', 'option_collection_id',
+                                               'parent_signature__signature_hash'))
+
+        signature_ids = [item['id'] for item in list(self.queryset)]
+
+        data = (PerformanceDatum.objects.select_related('push', 'repository')
+                                .filter(signature_id__in=signature_ids, repository__name=repository))
+
+        if revision:
+            data = data.filter(push__revision=revision)
+        else:
+            data = data.filter(push_timestamp__gt=startday, push_timestamp__lt=endday)
+
+        values = data.values_list('signature_id', 'value')
```

Comment:
I would push this down to line 483, or maybe just make it part of the `for (signature_id, value) in ...:` clause